build_tools/benchmarks/reporting/parse_shark_benchmarks.py

```diff
@@ -0,0 +1,305 @@
+
+  return st.to_html() + '<br/>'
+
+
+def generate_table(iree_csv,
+                   baseline_csv,
+                   shark_csv=None,
+                   shape_type='static',
+                   device='cpu',
+                   title='Benchmarks'):
+  """Generates a table comparing latencies between IREE, SHARK and a baseline.
+
+  Args:
+    iree_csv: Path to the csv file containing IREE latencies.
+    baseline_csv: Path to the csv file containing baseline latencies.
+    shark_csv: Path to the csv file containing SHARK-Runtime latencies. This is optional.
+    shape_type: Currently either `static` or `dynamic`.
+    device: Device used to run the benchmarks.
+    title: The title of the generated table.
+
+  Returns:
+    A HTML file containing the summarized report.
+  """
+  shark_df = None
+  if shark_csv is not None:
+    shark_df = pd.read_csv(shark_csv)
+    shark_df = shark_df.loc[(shark_df.shape_type == shape_type) &
+                            (shark_df.device == device)]
+
+  iree_df = pd.read_csv(iree_csv)
+  iree_df = iree_df.loc[(iree_df.shape_type == shape_type) &
+                        (iree_df.device == device)]
+
+  baseline_df = pd.read_csv(baseline_csv)
+  baseline_df = baseline_df.loc[(baseline_df.shape_type == shape_type) &
+                                (baseline_df.device == device)]
+
+  return _generate_table(iree_df, shark_df, baseline_df, title)
+
+
+def main(args):
+  """Summarizes benchmark results generated by the SHARK Tank."""
+  verison_html = '<i>IREE version: ' + args.iree_version + '</i><br/>'
+  verison_html = verison_html + '<i>SHARK version: ' + args.shark_version + '</i><br/>'
+  verison_html = verison_html + '<i>last updated: ' + datetime.today().strftime(
+      '%Y-%m-%d') + '</i><br/><br/>'
+  html = generate_header_and_legend(verison_html)
+
+  # Generate Server CPU Static.
+  html = html + generate_table(
+      args.cpu_iree_csv,
+      args.cpu_baseline_csv,
+      shark_csv=args.cpu_shark_csv,
+      shape_type='static',
+      device='cpu',
+      title='Server Intel Ice Lake CPU (Static Shapes)')
+
+  # Generate Server GPU Static.
+  html = html + generate_table(
+      args.gpu_iree_csv,
+      args.gpu_baseline_csv,
+      shark_csv=args.gpu_shark_csv,
+      shape_type='static',
+      device='cuda',
+      title='Server NVIDIA Tesla A100 GPU (Static Shapes)')
+
+  # Generate Server CPU Dynamic.
+  html = html + generate_table(
+      args.cpu_iree_csv,
+      args.cpu_baseline_csv,
+      shark_csv=args.cpu_shark_csv,
+      shape_type='dynamic',
+      device='cpu',
+      title='Server Intel Ice Lake CPU (Dynamic Shapes)')
+
+  # Generate Server GPU Dynamic.
+  html = html + generate_table(
+      args.gpu_iree_csv,
+      args.gpu_baseline_csv,
+      shark_csv=args.gpu_shark_csv,
+      shape_type='dynamic',
+      device='cuda',
+      title='Server NVIDIA Tesla A100 GPU (Dynamic Shapes)')
+
+  f = open(args.output_path, 'w')
+  f.write(html)
+  f.close()
```

Comment:
Recommend either using a context manager (the long-recommended way)

```suggestion
  with open(args.output_path, 'w') as f:
    f.write(html)
```

or using the newer pathlib way: https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text

```suggestion
  args.output_path.write_text(html)
```

(and separately use pathlib for the argument, suggestion below)

Reply:
Thanks for the tip. Changed to the newer pathlib way.